### PR TITLE
Feature/reusable datatable

### PR DESCRIPTION
**Pull request by sherviniv**
#### Please check if the PR fulfills these requirements

- [ ] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 

---

#### What kind of change does this PR introduce? (bug fix, feature, docs update, ...)

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### What is the new behavior? (if this is a feature change)
A table with call-back functions, pagination, sorting, searching, and dynamic cell rendering.

#### Other information
.eslintrc.js:
` 'linebreak-style': ['error', process.platform === 'win32' ? 'windows' : 'unix'],`
The configuration sets the expected line endings to be 'windows' if the operating system is Windows, and 'unix' if the operating system is not Windows
`'react/no-unstable-nested-components': 'off',`
For dynamic component rendering inside another one.

**ClientsPage has been implemented in order to show Table usage**

#### Please test your work using the [Accessibility Insights for Web](https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni) plugin. Include any relevant notes and screenshots.

---

#### TODO:
Use consistent styling for all the project styles

### CODE REVIEW

File: src/Components/DataTable/pagination.jsx
Comment (tackc) [inline src/Components/DataTable/pagination.jsx:1]: This looks great! Can we implement this reusable DataTable for the DonorsPage also?

Comment (tackc) [reply src/Components/DataTable/pagination.jsx:1]: Can we also combine this pagination.jsx sub-component with the `Paginator` component already in the repo

File: src/Pages/ClientsPage/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    try {
+      const response = await GetClients(currentPage, defaultPageSize);
+      setItemsCount(response.pagy.count);
+      setClients(response.data);
+    } catch (error) {
+      setItemsCount(0);
+      setClients([]);
+    }
+
+    // let data = mockData.sort((a, b) => {
```
Comment (renburnett) [inline src/Pages/ClientsPage/index.js:25]: @sherviniv no big deal, but could you please delete this commented out code, thanks!

File: src/Pages/ClientsPage/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import { DataTable, Pagination } from '../../Components/DataTable';
+import Navbar from '../../Components/Navbar';
+import Search from '../../Components/Search';
+import { GetClients } from '../../Services/ClientsService';
+
+function ClientsPage() {
+  const defaultPageSize = 8;
+  const [clients, setClients] = useState([]);
+  const [sortColumn, setSortColumn] = useState({ path: 'no', order: 'asc' });
```
Comment (renburnett) [inline src/Pages/ClientsPage/index.js:10]: @sherviniv no biggie but could you please change this to reflect what the variable names will be once this is merged

should be  `order` and `sort_by`

see open sort PR here:
https://github.com/FoodIsLifeBGP/banana-rails/pull/168/files#r1375314653

File: src/Pages/AllDonorsPage/index.js [hunk 3]
```diff
@@ -1,126 +1,87 @@
...
+  const getDonors = async () => {
+    try {
+      const response = await GetDonors(currentPage, defaultPageSize);
+      setDonors(response);
+    } catch (error) {
+      setItemsCount(0);
+      setDonors([]);
     }
+
+    // let data = mockData.sort((a, b) => {
```
Comment (renburnett) [inline src/Pages/AllDonorsPage/index.js:26]: please remove

File: src/Pages/DonorPage/index.js [hunk 4]
```diff
@@ -1,126 +1,87 @@
...
+  const getDonors = async () => {
+    try {
+      const response = await GetDonors(currentPage, defaultPageSize);
+      setDonors(response);
+    } catch (error) {
+      setItemsCount(0);
+      setDonors([]);
     }
+
+    // let data = mockData.sort((a, b) => {
```
Comment (renburnett) [inline src/Pages/DonorPage/index.js:26]: please remove

File: src/Pages/DonorPage/index.js [hunk 5]
```diff
@@ -1,126 +1,87 @@
...
-];
+import { GetDonors } from '../../Services/DonorsService';
+import { DataTable, Pagination } from '../../Components/DataTable';
 
 function DonorPage() {
-  const entriesPerPage = 10;
-  const [displayData, setDisplayData] = useState([]);
+  const defaultPageSize = 8;
+  const [donors, setDonors] = useState([]);
+  const [sortColumn, setSortColumn] = useState({ path: 'no', order: 'asc' });
```
Comment (renburnett) [inline src/Pages/DonorPage/index.js:12]: @sherviniv could you please change this to reflect what the variable names will be 
should be  `order` and `sort_by`

see open sort PR here:
https://github.com/FoodIsLifeBGP/banana-rails/pull/168/files#r1375314653

**Review by renburnett — CHANGES_REQUESTED** [review]
looks great just a few small tweaks :)

**Review by renburnett — APPROVED** [review]
looks great ! 🚀

**Pull request merged**